docs/Quick-start-and-introduction.md

```diff
@@ -138,19 +138,19 @@ $result = $request->getAttribute('authentication')->getResult();
 if ($result->isValid()) {
     $user = $request->getAttribute('identity');
 } else {
-    $this->log($result->getCode());
+    $this->log($result->getStatus());
     $this->log($result->getErrors());
 }
 ```
 
-The result sets objects code returned from `getCode()` will match one of these codes. The codes are implemented as class constants in the Result object.
+The result sets objects status returned from `getStatus()` will match one of these these constants in the Result object:
 
- * **1**: Success (Result::SUCCESS)
- * **0**: Failure (Result::FAILURE)
- * **-1**: Identify not found (Result::FAILURE_IDENTITY_NOT_FOUND)
- * **-2**: Credentials invalid (Result::FAILURE_CREDENTIAL_INVALID)
- * **-3**: Any other kind of failure (Result::FAILURE_OTHER)
- * **-4**: Credentials not found (Result::FAILURE_CREDENTIALS_NOT_FOUND)
+* `ResultInterface::SUCCESS`, when successful.
+* `ResultInterface::FAILURE`, on any general failure.
+* `ResultInterface::FAILURE_IDENTITY_NOT_FOUND`, when identity could not be found.
+* `ResultInterface::FAILURE_CREDENTIALS_INVALID`, when credentials are invalid.
+* `ResultInterface::FAILURE_CREDENTIALS_NOT_IN_REQUEST`, when credentials are missing in the request.
```

Comment:
I don't think we need to separate 'not found' and 'credentials not in request' cases. They are both not found errors.

Reply:
@markstory are you talking about `FAILURE_IDENTITY_NOT_FOUND` and `FAILURE_CREDENTIALS_NOT_IN_REQUEST`?

Reply:
@markstory it is not the same afaik. Identity not found means there was no match because the given identity was not found in the identity data source whereas not in request means the requesting endpoint/user forgot/omitted to supply the required credentials.

Reply:
@robertpustulka I am talking about those.

@inoas The credentials not existing is ~= to empty values which would result in the identity not being found. I'm not clear on *why* the separation is necessary.

Reply:
Well I didn't write that code so I am not sure about its intentions:

Here are 2 snips of code maybe the primary author of those code parts can explain better why both are required (or that one can be discaded):

Keep in mind that I renamed FAILURE_CREDENTIALS_NOT_IN_REQUEST to FAILURE_CREDENTIALS_MISSING:

In HttpDigestAuthenticator.php
```php
    public function authenticate(ServerRequestInterface $request, ResponseInterface $response)
    {
        $digest = $this->_getDigest($request);
        if (empty($digest)) {
            return new Result(null, Result::FAILURE_OTHER);
        }

        $user = $this->_identifier->identify([
            IdentifierInterface::CREDENTIAL_USERNAME => $digest['username']
        ]);

        if (empty($user)) {
            return new Result(null, Result::FAILURE_IDENTITY_NOT_FOUND);
        }

        if (!$this->validNonce($digest['nonce'])) {
            return new Result(null, Result::FAILURE_CREDENTIALS_INVALID);
        }

//...

        return new Result(null, Result::FAILURE_CREDENTIALS_INVALID);
```

In JwtAuthenticator.php:
```php
    public function authenticate(ServerRequestInterface $request, ResponseInterface $response)
    {
        try {
            $result = $this->getPayload($request);
        } catch (Exception $e) {
            return new Result(
                null,
                Result::FAILURE_CREDENTIALS_INVALID,
                [
                    'message' => $e->getMessage(),
                    'exception' => $e
                ]
            );
        }

        if (!$result instanceof stdClass) {
            return new Result(null, Result::FAILURE_CREDENTIALS_INVALID);
        }

        $result = json_decode(json_encode($result), true);

        $key = IdentifierInterface::CREDENTIAL_JWT_SUBJECT;
        if (empty($result[$key])) {
            return new Result(null, Result::FAILURE_CREDENTIALS_MISSING);
        }

//...

        if (empty($user)) {
            return new Result(null, Result::FAILURE_IDENTITY_NOT_FOUND, $this->_identifier->getErrors());
        }

        return new Result($user, Result::SUCCESS);
    }
```